File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int a =0,b=1,ans=0 , n=prices.size();
        while(b<n){
            if(prices[a]<prices[b]){
                ans = max(ans,(prices[b]-prices[a]));
                b++;
            }
            else{
                a=b;
                b++;
            }
        }
        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        a = 0
        b = 1
        n = len(prices)
        ans = 0
        while(b<n):
            if prices[a]<prices[b]:
                ans = max(ans,prices[b]-prices[a])
            else:
                a=b
            b+=1
        return ans
